File: source_files/Helper_scripts/upload_devices.py
import firebase_admin
from firebase_admin import credentials, firestore
import pandas as pd

# Загрузка данных из файла CSV
data = pd.read_csv('Упаковщики 1.csv')

# Инициализация Firebase Admin SDK
cred = credentials.Certificate('firebase-config.json')
firebase_admin.initialize_app(cred)

# Получение ссылки на Firestore
db = firestore.client()

# Функция для загрузки данных
def upload_devices(data):
    for index, row in data.iterrows():
        device_data = {
            'model': row['Модель'],
            'type': row['Тип'],
            'factory_serial_number': row['Заводской номер'],
            'region': row['Регион']
        }
        # Используем системный номер как идентификатор документа
        db.collection('Devices').document(row['Системный номер']).set(device_data)
        print(f"Uploaded device with system number {row['Системный номер']}")

# Загрузка данных
upload_devices(data)

File: source_files/Helper_scripts/uploadDevices.js
const admin = require('firebase-admin');
const csv = require('csv-parser');
const fs = require('fs');

// Инициализация Firestore
admin.initializeApp({
  credential: admin.credential.applicationDefault()
});

const db = admin.firestore();

// Путь к CSV файлу
const filePath = 'D:/firestore-setup/csv_ready/deep2241.csv';

fs.createReadStream(filePath)
  .pipe(csv({ separator: ';' })) // Используйте правильный разделитель
  .on('data', async (row) => {
    try {
      const id = row['id'];
      const type = row['type'];
      const model = row['model'];
      const SN = row['SN'];
      const region = row['region'];

      if (!id) {
        throw new Error('Missing id');
      }

      await db.collection('devices').doc(id).set({
        type,
        model,
        SN,
        region
      });
      console.log(`Added device with ID: ${id}`);
    } catch (error) {
      console.error(`Error adding device ${row['id']}: `, error);
    }
  })
  .on('end', () => {
    console.log('CSV file successfully processed');
  });
